File: DSA1/Reverse Nodes in k-Group.cpp
class Solution {
public:
    ListNode* reverseKGroup(ListNode* head, int k) {
        if(!head || !head->next || k == 1){
            return head;
        }
        int count = 0;
        ListNode* curr = head;
        while(curr){
            count++;
            curr = curr->next;
        }
        ListNode* prev = NULL;
        ListNode* next = NULL;
        ListNode* newHead = NULL;
        ListNode* x = NULL;
        ListNode* y;
        curr = head;
        while(count>=k){
            for(int i = 0; i<k; i++){
                next = curr->next;
                curr->next = prev;
                prev = curr;
                curr = next;
            }
            if(!newHead){
                newHead = prev;
            }
            if(x){
                x->next = prev;
            }
            y->next = curr;
            x = y;
            y = curr;
            prev = NULL;
            count -= k;
        }
        return newHead;
    }
    
};

File: DSA1/Reverse Nodes in k-Group.py
class Solution:
    def reverseKGroup(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        first,last,cur=head,head,head
        size=0
        while cur:
            size+=1
            cur=cur.next
            
        while last:
            a=[]
            if size>=k:
                for i in range(k):
                    a.append(last.val)
                    last=last.next

                for i in range(k):
                    first.val=a.pop()
                    first=first.next
            else:
                break
            size-=k
                
        return head
